### PR TITLE
M1 unit test

### PR DESCRIPTION
**Pull request by Sidu28**

### CODE REVIEW

File: src/test/EigenPod.t.sol [hunk 1]
```diff
@@ -533,6 +533,47 @@ contract EigenPodTests is ProofParsing, EigenPodPausingConstants {
...
+        eigenPodManager.pause(2 ** PAUSED_NEW_EIGENPODS);
+        cheats.stopPrank();
+
+        cheats.startPrank(podOwner);
+        cheats.expectRevert(bytes("Pausable: index is paused"));
+        eigenPodManager.createPod();
+        cheats.stopPrank();
+    }
+
+    function testDeployEigenPodFromNonPodManagerAddress(address nonPodManager) external fuzzedAddress(nonPodManager){
```
Comment (gpsanant) [inline src/test/EigenPod.t.sol:548]: testStakeOnEigenPodFromNonPodManagerAddress

**Review by gpsanant — APPROVED** [review]
:+1:

File: src/test/unit/StrategyBaseTVLLimitsUnit.sol [hunk 2]
```diff
@@ -57,8 +58,19 @@ contract StrategyBaseTVLLimitsUnitTests is StrategyBaseUnitTests {
         cheats.stopPrank();
     }
 
+    function testSetInvalidMaxPerDepositAndMaxDeposits(uint256 maxDepositsFuzzedInput, uint256 maxPerDepositFuzzedInput) public {
+        cheats.assume(maxDepositsFuzzedInput > 0);
+        cheats.assume(maxPerDepositFuzzedInput > 0);
```
Comment (ChaoticWalrus) [inline src/test/unit/StrategyBaseTVLLimitsUnit.sol:63]: these `assume`s seem unnecessary

File: src/test/unit/StrategyBaseTVLLimitsUnit.sol [hunk 3]
```diff
@@ -116,10 +128,43 @@ contract StrategyBaseTVLLimitsUnitTests is StrategyBaseUnitTests {
         require(strategyWithTVLLimits.totalShares() == depositAmount + sharesBefore, "total shares not updated correctly");
     }
 
+    function testDepositTVLLimit_ThenChangeTVLLimit(uint256 maxDeposits, uint256 newMaxDeposits) public {
+        cheats.assume(maxDeposits > 0);
```
Comment (ChaoticWalrus) [inline src/test/unit/StrategyBaseTVLLimitsUnit.sol:132]: again this actually seems unnecessary?

File: src/test/unit/StrategyManagerUnit.t.sol [hunk 4]
```diff
@@ -691,6 +704,48 @@ contract StrategyManagerUnitTests is Test, Utils {
         strategyManager.queueWithdrawal(strategyIndexes, strategyArray, shareAmounts, address(this), undelegateIfPossible);
     }
 
+    function testQueueWithdrawaMismatchedIndexAndStrategyArrayLength() external {
```
[the remark this answers is not in the document]
Comment (ChaoticWalrus) [reply src/test/unit/StrategyManagerUnit.t.sol]: fixed https://github.com/Layr-Labs/eigenlayer-contracts/pull/45/commits/86929bfe8f1abf6bc64fe45947bd848f798e666e

File: src/test/unit/StrategyBaseTVLLimitsUnit.sol (hunk 3 above)
Comment (ChaoticWalrus) [reply src/test/unit/StrategyBaseTVLLimitsUnit.sol]: fixed https://github.com/Layr-Labs/eigenlayer-contracts/pull/45/commits/09fd1573545db8d448b6e2f64f9c07da4795c137

File: src/test/unit/StrategyBaseTVLLimitsUnit.sol (hunk 2 above)
Comment (ChaoticWalrus) [reply src/test/unit/StrategyBaseTVLLimitsUnit.sol]: fixed https://github.com/Layr-Labs/eigenlayer-contracts/pull/45/commits/09fd1573545db8d448b6e2f64f9c07da4795c137

File: src/test/EigenPod.t.sol (hunk 1 above)
Comment (ChaoticWalrus) [reply src/test/EigenPod.t.sol]: fixed https://github.com/Layr-Labs/eigenlayer-contracts/pull/45/commits/41faecdd6aa82bfc541a31417c2877948a42a2ec

**Pull request merged**